src/Stack/Package.hs

```diff
@@ -803,12 +803,22 @@ resolveComponentFiles component build names = do
 
 -- | Get all C sources and extra source files in a build.
 buildOtherSources :: BuildInfo -> RIO Ctx [DotCabalPath]
-buildOtherSources build =
-    do csources <- liftM (map DotCabalCFilePath)
-                       (mapMaybeM resolveFileOrWarn (cSources build))
-       jsources <- liftM (map DotCabalFilePath)
-                       (mapMaybeM resolveFileOrWarn (targetJsSources build))
-       return (csources <> jsources)
+buildOtherSources build = do
+    cwd <- liftIO getCurrentDir
+    dir <- asks (parent . ctxFile)
+    file <- asks ctxFile
+    csources <-
+        forMaybeM (cSources build) $ \fp -> do
+            result <- resolveDirFile dir fp
+            case result of
+              Nothing -> do
+                warnMissingFile "File" cwd fp file
+                return Nothing
+              Just p -> return $ Just (DotCabalCFilePath p)
+    jsources <-
+        forMaybeM (targetJsSources build) $ \fp ->
```

Comment:
Does it make sense to use `findCandidate` for the Javascript source? If so, why? Why not use the same logic as the C sources?

Reply:
oh, that's just a stupid miss from me, will fix that